controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -99,11 +137,481 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases(ctx context.Contex
 }
 
 func (r *StorageClassClaimReconciler) reconcileProviderPhases(ctx context.Context, storagecluster *v1.StorageCluster) (reconcile.Result, error) {
-	log := ctrllog.FromContext(ctx)
-	log.Info("Running StorageClassClaim controller in Converged/Provider Mode")
+	if !r.storageConsumer.Spec.Enable {
+		r.storageClassClaim.Status.Phase = string(v1alpha1.StorageClassClaimStateDisabled)
+		return reconcile.Result{}, nil
+	}
```

Comment:
When is consumer disabled? Why do we report claim as disabled? It seems misleading.

Reply:
removed as this is not required to claim, it is required to consumer controller as we want to ack before starting reconciliation.